SMLHelper/Options/ModConfig.cs

```diff
@@ -0,0 +1,226 @@
+                    // and exactly what file is causing the error, with an option to open a JSON validator in
+                    // browser to help them fix the problem.
+                    // We'd also want this issue to stop QMods from considering the mod loaded properly, or
+                    // expose a method for mods to check if any mods had config file errors, so that mods
+                    // which rely on all mods loading properly before performing actions can act accordingly.
+                    // For now, without copying QMM's dialog class into SML, this is the best we can do.
+                    Logger.Announce($"Could not parse JSON file: {path}", LogLevel.Error, true);
+                    throw ex;
+                }
+            }
+            else if (saveDefaultConfigIfNotExist)
+            {
+                config.Save();
+            }
+            return config;
+        }
+
+        /// <summary>
+        /// Loads a given <see cref="ModConfig"/>'s options from the JSON file on disk and populates it.
+        /// </summary>
+        /// <typeparam name="T">The type of <see cref="ModConfig"/> to use for deserialization.</typeparam>
+        /// <param name="saveDefaultConfigIfNotExist">Whether a config file containing default values should
+        /// be created if it does not already exist.</param>
+        /// <returns>A <seealso cref="ModConfig"/> with its properties and fields populated from the 
+        /// associated JSON file.</returns>
+        /// <example>
+        /// <code>
+        /// using System;
+        /// using SMLHelper.V2.Options;
+        /// using SMLHelper.V2.Utility;
+        /// using UnityEngine;
+        /// 
+        /// public class Config : ModConfig {
+        ///     public KeyCode ActivationKey { get; set; } = KeyCode.Backspace;
+        ///     public Config() : base(fileName: "config") { }
+        /// }
+        /// 
+        /// public class MyMod {
+        ///     public static Config Config = ModConfig.Load&lt;Config&gt;();
+        ///     public static void Initialize() {
+        ///         string activationKey = KeyCodeUtils.KeyCodeToString(Config.ActivationKey);
+        ///         Console.WriteLine($"[MyMod] LOADED: ActivationKey = {activationKey}");
+        ///         // Will print "[MyMod] LOADED: ActivationKey = Backspace" in Player.log on first run,
+        ///         // and whatever value they have saved in their config.json for the "ActivationKey"
+        ///         // property on subsequent runs.
+        ///     }
+        /// }
+        /// </code>
+        /// </example>
+        /// <seealso cref="Load{T}(T, bool)"/>
+        /// <seealso cref="Load(bool)"/>
+        /// <seealso cref="Save"/>
+        public static T Load<T>(bool saveDefaultConfigIfNotExist = true) where T : ModConfig
+            => Load(Activator.CreateInstance<T>(), saveDefaultConfigIfNotExist);
```

Comment:
Add this to the constraints
```CS
where T : ModConfig, new()
```
This enforces the requirement that the class type `T` specified must have a parameterless constructor.

Reply:
Thanks, I didn't know about this.